core/src/main/java/org/apache/carbondata/core/datamap/DistributableDataMapFormat.java

```diff
@@ -87,37 +122,52 @@
+        List<Segment> segmentsToLoad = new ArrayList<>();
+        segmentsToLoad.add(distributable.getDistributable().getSegment());
         if (isJobToClearDataMaps) {
           // if job is to clear datamaps just clear datamaps from cache and return
           DataMapStoreManager.getInstance()
               .clearDataMaps(table.getCarbonTableIdentifier().getTableUniqueName());
           // clear the segment properties cache from executor
           SegmentPropertiesAndSchemaHolder.getInstance()
               .invalidate(table.getAbsoluteTableIdentifier());
-          blockletIterator = Collections.emptyIterator();
+          List<ExtendedBlocklet> list = new ArrayList<ExtendedBlocklet>();
+          list.add(new ExtendedBlocklet());
+          blockletIterator = list.iterator();
           return;
+        } else if (invalidSegments.size() > 0) {
+          // clear the segmentMap and from cache in executor when there are invalid segments
+          DataMapStoreManager.getInstance().clearInvalidSegments(table, invalidSegments);
         }
-        dataMaps = tableDataMap.getTableDataMaps(distributable.getDistributable());
-        List<ExtendedBlocklet> blocklets = tableDataMap
-            .prune(dataMaps,
-                distributable.getDistributable(),
+        SegmentUpdateStatusManager updateStatusManager = new SegmentUpdateStatusManager(table);
+        DataMapStoreManager.getInstance()
+            .refreshSegmentCacheIfRequired(table, updateStatusManager, segmentsToLoad);
+        List<ExtendedBlocklet> blocklets = new ArrayList<>();
+        if (dataMapExprWrapper instanceof DataMapExprWrapperImpl) {
+          TableDataMap defaultDataMap = DataMapStoreManager.getInstance()
+              .getDataMap(table, distributable.getDistributable().getDataMapSchema());
+          dataMaps = defaultDataMap.getTableDataMaps(distributable.getDistributable());
+          if (table.isTransactionalTable()) {
+            blocklets = defaultDataMap.prune(dataMaps, distributable.getDistributable(),
                 dataMapExprWrapper.getFilterResolverIntf(distributable.getUniqueId()), partitions);
-        for (ExtendedBlocklet blocklet : blocklets) {
-          blocklet.getDetailInfo();
-          blocklet.setDataMapUniqueId(distributable.getUniqueId());
+          } else {
+            blocklets = defaultDataMap.prune(segmentsToLoad, filterResolverIntf, partitions);
+          }
+          blocklets = DataMapUtil
+              .pruneDataMaps(table, filterResolverIntf, segmentsToLoad, partitions, blocklets);
+        } else {
+          blocklets = DataMapUtil
+              .pruneDataMaps(table, filterResolverIntf, segmentsToLoad, partitions, blocklets,
+                  dataMapLevel);
         }
         blockletIterator = blocklets.iterator();
       }
 
-      @Override
-      public boolean nextKeyValue() throws IOException, InterruptedException {
+      @Override public boolean nextKeyValue() throws IOException, InterruptedException {
```

Comment:
why changing the function signature in this class?

Reply:
The defined core formatted for carbon puts the @Override in the same line as the signature.